netket/graph/lattice.py

```diff
@@ -236,7 +151,8 @@ def __init__(
         atoms_coord: Optional[_np.ndarray] = None,
         distance_atol: float = 1e-5,
         point_group: Optional[PointGroup] = None,
-        max_neighbor_order: int = 1,
+        max_neighbor_order: Optional[int] = None,
```

Comment:
Why do we need this to be `Optional[int] = None` now instead of just `1`, which according to the docstring is the default?

Reply:
Because if you specify `custom_edges`, it would look like you specified both arguments, which should error out, even though you didn't want specify `max_neighbor_order`.